### PR TITLE
[@mantine/rte] Add mentions with quill-mention

### PR DESCRIPTION
**Pull request by hammo92**
PR to add support for mentions through quill-mention. 

Adds a mentionGroups props to allow different mention lists for different triggers.

usage:
carbon

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
     "dayjs": "^1.10.5",
     "html-react-parser": "^1.3.0",
     "prism-react-renderer": "^1.2.1",
+    "quill-mention": "^3.0.8",
```
Comment (rtivital) [inline package.json:136]: Add quill-mention as a dependency in `@mantine/rte` package

File: src/mantine-rte/src/components/RichTextEditor/RichTextEditor.tsx [hunk 2]
```diff
@@ -54,6 +76,9 @@ export interface RichTextEditorProps
   /** Make toolbar sticky */
   sticky?: boolean;
 
+  /** list of mentionGroupss divided into groups */
```
Comment (rtivital) [inline src/mantine-rte/src/components/RichTextEditor/RichTextEditor.tsx:79]: **mentionGroupss** typo

File: src/mantine-rte/src/components/RichTextEditor/RichTextEditor.tsx [hunk 3]
```diff
@@ -89,6 +115,27 @@ export const RichTextEditor = forwardRef<Editor, RichTextEditorProps>(
         imageUploader: {
           upload: (file: File) => onImageUpload(file),
         },
+        mention: {
+          allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
+          //spaceAfterInsert: false,
```
Comment (rtivital) [inline src/mantine-rte/src/components/RichTextEditor/RichTextEditor.tsx:120]: Remove commented out code

**Review by rtivital — COMMENTED** [review]
@hammo92 please resolve issues in code review and add documentation with a demo to merge this PR

**Comment by hammo92** [comment]
@rtivital does yarn only have to be run at the root?

**Comment by rtivital** [comment]
Alright, thanks, I'll handle yarn errors on my own

**Pull request merged**